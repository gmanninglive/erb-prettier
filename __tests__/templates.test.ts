import fs from "fs";
import { format } from "prettier";

describe("e2e", () => {
  it("formats template files", () => {
    const root = "__tests__/templates";
    const paths = fs.readdirSync(root);

    paths
      .filter((p) => !p.includes("prettier"))
      .forEach((path) => {
        const input = fs.readFileSync(root.concat("/", path)).toString();

        const output = format(input, {
          parser: "erb",
          pluginSearchDirs: [".."],
          plugins: ["erb-prettier"],
        });

        expect(output).toMatchSnapshot(path);
        fs.writeFileSync(
          [
            "__tests__",
            "__snapshots__",
            path.replace("html.erb", "prettier.html.erb"),
          ].join("/"),
          output
        );
      });
  });
});
