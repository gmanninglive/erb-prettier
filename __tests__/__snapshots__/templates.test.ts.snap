// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e formats template files: _header.html.erb 1`] = `
"<header
  class="w-full border-b border-gray-dark flex flex-col lg:flex-row md:justify-center lg:justify-between"
>
  <div class="lg:w-1/2 lg:flex lg:pr-2.5 mb-2.5">
    <button
      class="hidden lg:block px-7.5 h-[35px] mr-2.75 bg-blue-secondary hover:bg-blue-hovef disabled:bg-gray-dark disabled:hover:bg-gray-dark text-white text-lg font-normal tracking-tighter rounded-md"
      id="submit"
      onclick='$("#form_submit").trigger("click")'
    >
      Export
    </button>
    <%= render partial: 'search_form' %>
  </div>
  <div
    class="mt-3 inline-flex gap-x-0.5 overflow-x-auto overflow-y-hidden whitespace-nowrap lg:overflow-visible"
  >
    <div>
      <div>
        <%= render partial: 'header_tab',
            locals: { link_name: "All",
            link_path: customers_orders_path(query: { state: 'all' }),
            this_state: 'all' } %>
      </div>
    </div>

    <%= render partial: 'header_tab',
        locals: { link_name: "Enquiries/Booked",
        link_path: customers_orders_path(query: { state: 'booked' }),
        this_state: 'booked' } %>

    <%= render partial: 'header_tab',
        locals: { link_name: "Active Orders",
        link_path: customers_orders_path(query: { state: 'active' }),
        this_state: 'active' } %>

    <%= render partial: 'header_tab',
        locals: { link_name: "Completed Orders",
        link_path: customers_orders_path(query: { state: 'completed' }),
        this_state: 'completed' } %>
  </div>
</header>
"
`;
